src/main/java/com/guciowons/shoppingbasket/Basket/BasketController.java

```diff
@@ -19,11 +20,19 @@ public ResponseEntity<Basket> createBasket(){
     }
 
     @PutMapping(value = "/{basketId}/{productId}/{quantity}")
-    public ResponseEntity<BasketSummarized> addProductToBasket(@PathVariable int basketId, @PathVariable int productId, @PathVariable int quantity){
+    public ResponseEntity addProductToBasket(@PathVariable int basketId, @PathVariable int productId, @PathVariable int quantity){
         try{
-            return new ResponseEntity<>(basketService.addProductToBasket(basketId, productId, quantity), HttpStatus.ACCEPTED);
+            return ResponseEntity
+                    .status(HttpStatus.ACCEPTED)
+                    .body(basketService.addProductToBasket(basketId, productId, quantity));
         }catch(IllegalArgumentException e){
-            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+            return ResponseEntity
```

Comment:
Ogólnie spoko, ale patrząc na to, ile może być różnych typów błędów itd to warto czasem ogarnąć to w trochę inny sposób.

Popatrz na https://www.baeldung.com/exception-handling-for-rest-with-spring i na punkt 4.

Możesz zrobić jakiś ogólny typ błędów zwracany przez API (eg ApiException), potem błędy biznesowe mogą po nim dziedziczyć i możęsz napisać jeden exception handler na te błędy, który jak coś takiego się wydarzy to je złapie i zrobi z nich ładne błędy.

Nie musisz tego teraz poprawiać, ale warto sobie to ogarnąć.